### ISSUE TITLE
Account bug

### ISSUE THREAD

**Issue by para2022** [WWBN/AVideo#8790]
You have 'email' twice is the accounts causing problems with accounts. It should be user name i think but shows email instead.

**Comment by DanielnetoDotCom** [comment]
Can you please provide a screenshot?

**Comment by para2022** [comment]
What of users personal details for all to see?

**Comment by para2022** [comment]
So currently my name is there how it was before turning on email login. It used to say username there, and that username is still there. I cannot save the account page cos it brings me back to what used to be username and asks for email again. That is super confusing for users. It should still stay as username?

![Screenshot 2024-01-12 at 17 31 53](https://github.com/WWBN/AVideo/assets/107706817/8d5a62be-b2b3-40ee-b5df-0d46cf389f4f)

**Comment by DanielnetoDotCom** [comment]
I see, it is because your username is also the email. 

I will send you updates.

**Comment by para2022** [comment]
Thats much better thanks mate. Will everyones account be ok as they are though? Obviously if they go into their account it will prompt them to fix the email yea?

**Comment by DanielnetoDotCom** [comment]
Yes

**Comment by para2022** [comment]
ok thank you

**Comment by para2022** [comment]
I just spotted something on video owner showing email instead of name in video panel

![Screenshot 2024-01-12 at 19 38 06](https://github.com/WWBN/AVideo/assets/107706817/2974d4df-36f4-4f1d-84ba-c2c2873f511c)

**Comment by para2022** [comment]
Its not changed everyones only mine since i entered my email in what was the username box.

**Comment by DanielnetoDotCom** [comment]
Check what you have here 

![image](https://github.com/WWBN/AVideo/assets/18009293/0f421d3e-fbdc-4132-a40c-94f7baaf56cc)

**Comment by para2022** [comment]
also before i could either enter name or email address not it is only accepting email and not name. lol

**Comment by DanielnetoDotCom** [comment]
Yes, if you select it must be an email you need to make sure it is an email

**Comment by para2022** [comment]
i have 
Do NOT identify user by Username (selected)
Do NOT identify user by Email (selected)
Do NOT identify user by Name (unselected)

But it still shows email as video owner on upload

**Comment by DanielnetoDotCom** [comment]
On your front page, I can see this is showing correct. 

![image](https://github.com/WWBN/AVideo/assets/18009293/ae8df05f-e8b1-4981-8b0b-9c84d4cb2cde)

I do not see any problem with showing the email or username in the videos manager. 

only the video owner or admin can see it.

**Comment by para2022** [comment]
Its the video panel its affecting:


![Screenshot 2024-01-12 at 20 01 27](https://github.com/WWBN/AVideo/assets/107706817/80fcacfd-eec9-4b10-b58d-b646c5591907)

**Comment by DanielnetoDotCom** [comment]
Sorry, I am not understanding what is the issue or problem

**Comment by para2022** [comment]
The video owner is not their name now it is their email instead you can see here

![Screenshot 2024-01-12 at 20 04 28](https://github.com/WWBN/AVideo/assets/107706817/073cbe0c-8e4e-4eac-94ab-5318917dd1b3)

**Comment by para2022** [comment]
![Screenshot 2024-01-12 at 20 06 18](https://github.com/WWBN/AVideo/assets/107706817/13bf39e8-6ec4-44b0-a507-4932b587fda4)

**Comment by DanielnetoDotCom** [comment]
Once your users update their usernames, they will also show the email

**Comment by para2022** [comment]
But how would i know who these people are in the video panel? LOL i cant identify them by their emails

**Comment by DanielnetoDotCom** [comment]
you can search in the users manager right?

**Comment by para2022** [comment]
seems a ridiculous amount to work to look them up in the user database to see who they are.

**Comment by para2022** [comment]
Their must be a much easier way of simplifying the whole account system seems overly complicated for users

**Comment by DanielnetoDotCom** [comment]
I disagree that is too much work, you can search not in the database, but in the users manager.

but anyway. the owner is now clickable, and you will go to his channel

**Comment by GreenDragonTavern** [comment]
> also there is no username anymore, since you force everyone to use their email

Native login, isn't this just a simple flag to validate (login) by username or email address? Username should never disappear just because you configure AVideo to login by email. Are you replacing username with email in database under such a configuration?

**Comment by DanielnetoDotCom** [comment]
> > also there is no username anymore, since you force everyone to use their email
> 
> Native login, isn't this just a simple flag to validate (login) by username or email address? Username should never disappear just because you configure AVideo to login by email. Are you replacing username with email in database under such a configuration?

Correct!

**Comment by GreenDragonTavern** [comment]
That is a terrible design and eliminates option of returning back to username as login validation. Why in the world would you overwrite username with duplicate data (email) when a simple flag establishes login preference and logic handles the rest?

**Comment by DanielnetoDotCom** [comment]
The username is unique, the email is not mandatory to be unique. that is why I cannot use the email field

**Comment by GreenDragonTavern** [comment]
> The username is unique, the email is not mandatory to be unique. that is why I cannot use the email field

You are going from bad to worse. Your design dictates that email should not be an option regarding login, period end of story. You are overwriting username (which is still a poor design) with data that 1) isn't mandatory and 2) not unique.

**Comment by DanielnetoDotCom** [comment]
This is your opinion!! thanks anyway

**Comment by GreenDragonTavern** [comment]
Overwriting a primary key field with data that isn't mandatory is not opinion.

**Comment by DanielnetoDotCom** [comment]
Username is not primary, it is unique

**Comment by GreenDragonTavern** [comment]
Overwriting a unique key field with data that isn't mandatory is not opinion.

**Comment by DanielnetoDotCom** [comment]
OMG, I am not willing to explain

**Comment by GreenDragonTavern** [comment]
You don't have to. Understand exactly what AVideo does when enabling email validation. My suggestion is if you're going to allow login using email then the email field should be defined as mandatory and unique, no different than username. This way there is no writing over username with email, and logic dictates the database field to validate.

**Comment by para2022** [comment]
Hi greendragon i am guessing Dan has been willing to move forward on this because changing the whole system could break the site/plugins in other ways and he would have to re-write the whole code? Im not sure I'm not a coder on that scale. Dan has always from my experience fixed issues and for a single developer he  does great work even though sometimes I don't agree and get angry and frustrated. It is his project after all. I am not happy with the user system i do think it could be simplified. Name and username is very confusing. Name should be just part of the registarion but not part of the AVideo system Any system I have ever used uses username and password. Because of the way it is now if Dan was to change it then it could mess other users sites up? I dunno. i.e if he removed name and users front pages are using peoples names and not username then it could mess them up.

In an ideal world those 3 options in the custoamsdvanced should go and the system should be simplified to username and email without any need for additional options and that users can login with username or email. Im assuming Dan is been over cautious here. I don't even know if it's possible now. Maybe it would be if the AVIdeo script was only just released and not active on peoples sites. I don't know mate i'm just trying to play devils advocate here. I don't know what the solution is but all i know is it is quite messy and with AVideo being such an awesome platform its a shame it doesn't have a much more sensible login/registration system and certain aspects like this let the platform down and its a crying shame, but i do appreciate all his hard work. The forgot password function is a mess for instance. Nobody knows they have to enter their name and then press it. users are generally not very bright and you can relate to this and in development you always have to take that into account. Not everyone is as clued up as us, most are stupid. You should be able to press the forgot password and in its popup it asks for your email address there (not username) and enter the captcha there and that is more logical and simplistic. I get endless emails saying they cant understand it and I agree.

**Comment by GreenDragonTavern** [comment]
Hello @para2022 - In this instance Daniel's shorts got in a bunch for no reason. Users table has two unique keys defined (id as primary, user as unique). Based on design, user is the only sensible option to validate login (nobody is going to know their id number to login with). This is a guess but initial design of AVideo did not consider email as a login option and came about later on. The email field is not indexed, defaults to NULL and allows duplicates, thus eliminating its data as a login mechanism for any up and running AVideo system. 

Simply put, implement email as the login mechanism immediately after install and never look back, otherwise, never use email as login unless addicted to chaos 😉

**Comment by para2022** [comment]
I received this before. That user has never logged in. He must be confused with his username. Obviously the below is edited for security purposes.

A new user has signup
name: Rolando
user: his_email@gmail.com
captcha: (edited)
redirectUri: 
email: his_email@gmail.com
comment:

**Comment by GreenDragonTavern** [comment]
Unfortunately using email as login in AVideo is a kludge. Since you have already committed, a follow up email immediately after registration may be necessary clarifying login process and nomenclature.

**Comment by para2022** [comment]
Dan you know if i go to the admin user panel? If i change my username back from the email address to my name again then i can login as both username and email and if i upload then it shows my name again how it should be rather than the email address. How weird. But i cannot do it in the account though.

**Comment by DanielnetoDotCom** [comment]
@para2022 only admin can do it

**Comment by para2022** [comment]
ok thanks
